### PR TITLE
[PLAY-43] Add size prop to the button kit

### PR DESCRIPTION
**Pull request by JelenaD1**
This PR adds ability for developer to add a `size` prop to the button kit. Currently there's only one size(14px) set as default. With this change developer can add `sm` and `lg` size to the button kit.

#### Screens

Rails version: 
Playbook_Design_System
React version: 
Playbook_Design_System_and_jelenadurovic_Jelenas-MacBook-Pro___Desktop_Nitro_playbook_playbook_playbook

#### Runway Ticket URL
[https://nitro.powerhrg.com/runway/backlog_items/PLAY-43](https://nitro.powerhrg.com/runway/backlog_items/PLAY-43)

#### How to test this
Specs are covered. To test in a browser simply add a size prop to the button. 

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_button/_button.scss [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
+  "lg":   1.125rem,
+);
+
 [class^=pb_button_kit]{
   @include pb_button;
+  // Size =================
+  @each $name, $size in $pb_button_sizes {
+    &[class*=_#{$name}] {
+      font-size: $size;
+      padding: calc(#{$size} / 2) calc(#{$size} * 2.42) !important;
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_button/_button.scss:15]: This is very cool, but I do have just one question on the origin of `2.42` within the calculated x padding here. Is there any reference to point to? Mostly I'm just curious. On the `!important` part, what would make this necessary? Which other selector(s) are making this necessary?

File: playbook/app/pb_kits/playbook/pb_button/_button.scss [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+    }
+  }
+  &[class*=_sm] {
+    min-height: 0;
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_button/_button.scss:19]: Curious on this one too. Once the button is sized down to "sm", does this keep the box model consistent?

**Review by thestephenmarshall — COMMENTED** [review]
@JelenaD1 nice work so far. One thing I noticed along with my questions above was that the [Jest test for _button.jsx](https://github.com/powerhome/playbook/blob/master/playbook/app/pb_kits/playbook/pb_button/button.test.js) was not updated yet. Getting close!

File: playbook/app/pb_kits/playbook/pb_button/_button.scss (hunk 1 above)
Comment (JelenaD1) [reply playbook/app/pb_kits/playbook/pb_button/_button.scss:15]: Jon suggested to convert size values from px to rems so we can calculate the padding.  `!imprtant`is used to override the previous styles on this specific property.

File: playbook/app/pb_kits/playbook/pb_button/_button.scss (hunk 2 above)
Comment (JelenaD1) [reply playbook/app/pb_kits/playbook/pb_button/_button.scss:19]: Since sm size is only 12px min-height needs to be set to 0 only for this size.

**Comment by JelenaD1** [comment]
@thestephenmarshall I made a small change to React code to match with classname of Rails version so I need your re-approval sir.

**Review by kre8sions — APPROVED** [review]

**Pull request merged**